SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -21,62 +21,113 @@ SpatialLoadBalanceEnforcer::SpatialLoadBalanceEnforcer(const PhysicalWorkerName&
-		if (AuthorityIntentComponent == nullptr)
-		{
-			// TODO(zoning): There are still some entities being created without an authority intent component.
-			// For example, the Unreal created worker entities don't have one. Even though those won't be able to
-			// transition, we should have the intent component on them for completeness.
-		 	UE_LOG(LogSpatialLoadBalanceEnforcer, Log, TEXT("Requested authority change for entity without AuthorityIntent component. EntityId: %lld"), AuthOp.entity_id);
-			return;
-		}
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning,
+			TEXT("Load balancer notified of add component %d for entity %lld which is not a load balancing component"),
+			Op.data.component_id, Op.entity_id);
+		return;
+	}
 
-		const PhysicalWorkerName* OwningWorkerId = VirtualWorkerTranslator->GetPhysicalWorkerForVirtualWorker(AuthorityIntentComponent->VirtualWorkerId);
-		if (OwningWorkerId != nullptr &&
-			*OwningWorkerId == WorkerId &&
-			StaticComponentView->HasAuthority(AuthOp.entity_id, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Verbose, TEXT("No need to queue newly authoritative entity %lld because this worker is already authoritative."), AuthOp.entity_id);
-			return;
-		}
-		QueueAclAssignmentRequest(AuthOp.entity_id);
+	if (CanEnforce(Op.entity_id))
+	{
+		MaybeQueueAclAssignmentRequest(Op.entity_id);
+	}
+}
+
+void SpatialLoadBalanceEnforcer::OnLoadBalancingComponentRemoved(const Worker_RemoveComponentOp& Op)
+{
+	if (!(Op.component_id == SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID || Op.component_id == SpatialConstants::ENTITY_ACL_COMPONENT_ID))
+	{
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning,
+			TEXT("Load balancer notified of remove component %d for entity %lld which is not a load balancing component"),
+			Op.component_id, Op.entity_id);
+		return;
+	}
+
+	if (AclAssignmentRequestIsQueued(Op.entity_id))
+	{
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning,
+			TEXT("Component %d for entity %lld removed. Can no longer enforce the previous request for this entity."),
+			Op.component_id, Op.entity_id);
+		return;
+	}
+}
+
+void SpatialLoadBalanceEnforcer::OnEntityRemoved(const Worker_RemoveEntityOp& Op)
+{
+	if (AclAssignmentRequestIsQueued(Op.entity_id))
+	{
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning,
+			TEXT("Entity %lld removed. Can no longer enforce the previous request for this entity."), Op.entity_id);
+		return;
+	}
+}
+
+void SpatialLoadBalanceEnforcer::OnAclAuthorityChanged(const Worker_AuthorityChangeOp& AuthOp)
+{
+	if (AuthOp.component_id != SpatialConstants::ENTITY_ACL_COMPONENT_ID)
+	{
+		UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Loadbalancer informed of authority change for entity %lld that was not related to the ACL component."), AuthOp.entity_id);
+		return;
+	}
+
+	if (AclAssignmentRequestIsQueued(AuthOp.entity_id) && AuthOp.authority == WORKER_AUTHORITY_NOT_AUTHORITATIVE)
```

Comment:
NIT, am used to seeing checks for WORKER_AUTHORITY_AUTHORITATIVE and brain just assumed this, can we change to
```suggestion
	if (AclAssignmentRequestIsQueued(AuthOp.entity_id) && AuthOp.authority != WORKER_AUTHORITY_AUTHORITATIVE)
```

Reply:
Sure, makes it more readable, changed